firmware/steering/kia_soul_ps/steering_control_module.ino

```diff
@@ -743,100 +139,101 @@ void setup( )
-            disable_control( );
-        }
-        else if ( current_ctrl_state.control_enabled == true )
-        {
-
-/*******************************************************************************
-*   WARNING
-*
-*   The ranges selected to do steering control are carefully tested to
-*   ensure that a torque is not requested that the vehicles steering motor
-*   cannot handle. By changing any of this code you risk attempting to actuate
-*   a torque outside of the vehicles valid range. Actuating a torque outside of
-*   the vehicles valid range will, at best, cause the vehicle to go into an
-*   unrecoverable fault state. Clearing this fault state requires one of Kia's
-*   native diagnostics tools, and someone who knows how to clear DTC codes with
-*   said tool.
-*
-*   It is NOT recommended to modify any of the existing control ranges, or
-*   gains, without expert knowledge.
-*******************************************************************************/
-
-            // Calculate steering angle rates (degrees/microsecond)
-            double steering_angle_rate =
-                ( current_ctrl_state.current_steering_angle -
-                  current_ctrl_state.steering_angle_last ) / 0.05;
-
-            double steering_angle_rate_target =
-                ( current_ctrl_state.commanded_steering_angle -
-                  current_ctrl_state.current_steering_angle ) / 0.05;
-
-            // Save the angle for next iteration
-            current_ctrl_state.steering_angle_last =
-                current_ctrl_state.current_steering_angle;
-
-            steering_angle_rate_target =
-                constrain( ( double )steering_angle_rate_target,
-                           ( double )-current_ctrl_state.steering_angle_rate_max,
-                           ( double )current_ctrl_state.steering_angle_rate_max );
-
-            pid_params.derivative_gain = current_ctrl_state.SA_Kd;
-            pid_params.proportional_gain = current_ctrl_state.SA_Kp;
-            pid_params.integral_gain = current_ctrl_state.SA_Ki;
-
-            pid_update(
-                    &pid_params,
-                    steering_angle_rate_target,
-                    steering_angle_rate,
-                    0.050 );
-
-            double control = pid_params.control;
-
-            control = constrain( ( float ) control,
-                                 ( float ) -1500.0f,
-                                 ( float ) 1500.0f );
-
-            struct torque_spoof_t torque_spoof;
-
-            calculate_torque_spoof( control, &torque_spoof );
-
-            torque_sum = (uint8_t) ( torque_spoof.low + torque_spoof.high );
-
-            dac.outputA( torque_spoof.low );
-            dac.outputB( torque_spoof.high );
-        }
-        else
-        {
-            current_ctrl_state.override_flag.wheel = 0;
-
-            pid_zeroize( &pid_params, STEERING_WINDUP_GUARD );
-        }
-    }
+	if( test_code == false )
```

Comment:
This is probably ok for now, but you might want to consider have a compile-time switch to enable these tests.  This is adding overhead for every loop iteration that we might, at some point, not want to incur.